### PR TITLE
asynchttpserver: ipv6 support

### LINKED ISSUE #18705: asynchttpserver binds to IPv4 only
https://github.com/nim-lang/Nim/blob/901c5ded527bb06c52bfc1fd38c9a5fadee0f49a/lib/pure/asynchttpserver.nim#L381

### PR DESCRIPTION
**Pull request by ghost**
Fix #18705

### CODE REVIEW

File: lib/pure/asynchttpserver.nim [hunk 1]
```diff
@@ -404,6 +404,7 @@ proc acceptRequest*(server: AsyncHttpServer,
 proc serve*(server: AsyncHttpServer, port: Port,
             callback: proc (request: Request): Future[void] {.closure, gcsafe.},
             address = "";
+            domain: Domain = AF_INET;
             assumedDescriptorsPerRequest = -1) {.async.} =
```
Comment (ringabout) [inline lib/pure/asynchttpserver.nim]: Maybe you can make a new overload to maintain backward compatibility.

```nim
proc serve*(server: AsyncHttpServer, port: Port,
            callback: proc (request: Request): Future[void] {.closure, gcsafe.},
            address: string;
            domain: Domain;
            assumedDescriptorsPerRequest = -1) {.async.} =
```

Or just place `domain` parameter at the end(thought a bit ugly)
see also https://github.com/dom96/httpbeast/pull/45

File: lib/pure/asynchttpserver.nim [hunk 2]
```diff
@@ -404,6 +404,7 @@ proc acceptRequest*(server: AsyncHttpServer,
 proc serve*(server: AsyncHttpServer, port: Port,
             callback: proc (request: Request): Future[void] {.closure, gcsafe.},
             address = "";
+            domain: Domain = AF_INET;
```
Comment (Araq) [inline lib/pure/asynchttpserver.nim]: Move this parameter after `assumedDescriptorsPerRequest` please.

Comment (ghost) [reply lib/pure/asynchttpserver.nim]: Done

File: lib/pure/asynchttpserver.nim (hunk 1 above)
Comment (ghost) [reply lib/pure/asynchttpserver.nim]: Moved domain to the end.

File: tests/errmsgs/tgcsafety.nim [hunk 3]
```diff
@@ -6,7 +6,7 @@ tgcsafety.nim(31, 18) Error: type mismatch: got <AsyncHttpServer, Port, proc (re
 but expected one of:
 proc serve(server: AsyncHttpServer; port: Port;
            callback: proc (request: Request): Future[void] {.closure, gcsafe.};
-           address = ""; assumedDescriptorsPerRequest = -1): owned(
+           address = ""; assumedDescriptorsPerRequest = -1; domain: AF_INET): owned(
```
Comment (ringabout) [inline tests/errmsgs/tgcsafety.nim]:
```suggestion
           address = ""; assumedDescriptorsPerRequest = -1; domain = AF_INET): owned(
```

**Pull request merged**